components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useCallback, useEffect, useMemo, useState } from "react";
+import { Button } from "../../components/Button";
+import { useDownloadUsageCSV } from "./download-usage-csv";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+import { Dayjs } from "dayjs";
+import { useToast } from "../../components/toasts/Toasts";
+import { useCurrentOrg } from "../../data/organizations/orgs-query";
+import { ReactComponent as DownloadIcon } from "../../icons/Download.svg";
+import { ReactComponent as ExclamationIcon } from "../../images/exclamation.svg";
+import { LinkButton } from "../../components/LinkButton";
+import { saveAs } from "file-saver";
+import prettyBytes from "pretty-bytes";
+import { ProgressBar } from "../../components/ProgressBar";
+
+type Props = {
+    attributionId: AttributionId;
+    startDate: Dayjs;
+    endDate: Dayjs;
+};
+export const DownloadUsage: FC<Props> = ({ attributionId, startDate, endDate }) => {
+    const { data: org } = useCurrentOrg();
+    const { toast } = useToast();
+
+    const handleDownload = useCallback(async () => {
+        if (!org) {
+            return;
+        }
+
+        toast(
+            <DownloadUsageToast
+                orgName={org?.slug ?? org?.id}
+                attributionId={attributionId}
+                startDate={startDate}
+                endDate={endDate}
+            />,
+            {
+                autoHide: false,
+            },
+        );
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" onClick={handleDownload} className="flex flex-row" icon={<DownloadIcon />}>
+            Export as CSV
```

Comment:
Yah, what happens is the first one is doing the actual loading, and subsequent ones get to skip the download since it’s in memory already, so their progress bars go from 0 to done. It’s partially that way to protect us from hammering our api for the same export data if a user does it multiple times. I agree we could probably improve this, but I also think it’s a bit of an edge case (user hammering the export button). I do still like the toast for this though due to it’s more persistent non-blocking UI. 

One thought I had was we could try to limit to 1 active export download at a time, especially if it’s for the same date range as an active one. This comes with a decent amount of complexity though as a tradeoff.

Another would be to introduce a short disabled state to the export button once clicked. The export will take at least 1 second to transition to complete, so we could disable the button for 1 second. Not sure that’s worth it, but maybe?

Reply:
The boring (simple) solution to disable the button for 1 second sounds good. ✨ 

But yeah, easy to skip for now. Your call.